Add descriptive names for special values passed to stfl_run()

dennisschagt opened the pull request:
Relevant quotes from the [`stfl_run()` documentation](http://svn.clifford.at/stfl/trunk/README):
> Return the next event. If no more prior generated events are waiting display
> the form and process one input character. The event string can be an null
> value when something changed in the form (e.g. the user changed the focus of
> the current widget) but all inputs have been handled internally inside of STFL.

>The … parameter is a timeout in ms. When no key has been pressed until this
> timeout has been reached the function returns with a "TIMEOUT" event. Set this
> parameter to 0 to disable the timeout.
> 
> When the timeout parameter is set to -1 the form is displayed independent of
> the current status of the event queue and the function returns right after
> displaying the form without handling any input characters. In this mode always
> an null value is returned.
> 
> When the timeout parameter is set to -3, rendering (and setting the :x, :y, :w
> and :h pseudo-variables) is done but the screen is not updated and no events
> are fetched. This is useful for incrementing rendering processes where
> appropriate :x, :y, :w and/or :h values are needed for finishing the layout.

Minoru reviewed (commented):
This is a cool idea! The code is way more readable now.

I'm slightly worried that someone might re-introduce a call to `run()` later on; this is fairly unlikely as this is a low-level function with an obscure argument, but still. Are there any ways to make sure this won't ever happen? (I expect this to turn into a discussion, so marking this review as a "comment" for now.)

I also notice `PbView` class still contains two calls to `run()` — does something prevent us from changing them to the new API?

dennisschagt commented:
> I'm slightly worried that someone might re-introduce a call to `run()` later on; this is fairly unlikely as this is a low-level function with an obscure argument, but still. Are there any ways to make sure this won't ever happen?

I'm not sure there is a way to make it completely impossible but I did remove the `FormAction::get_form()` method to reduce the number of classes with access to `Stfl::Form::run()`.

> I also notice `PbView` class still contains two calls to `run()` — does something prevent us from changing them to the new API?

`PbView` uses `Stfl::Form` directly, without using the `FormAction` class.
The `FormAction` class now depends on Newsboat's `View` class so we will have to remove that dependency to make it usable in Podboat.

I would like to get Podboat's `PbView` and Newsboat's `View` a bit more consistent but that will take some time (and a different PR).

Minoru commented on src/dialogsformaction.cpp:
```diff
@@ -27,7 +27,7 @@ void DialogsFormAction::init()
 {
 	set_keymap_hints();
 
-	f.run(-3); // compute all widget dimensions
```
Out of curiosity: why all the methods in this PR are called through `this`? I know C++ sometimes needs this when the compiler can't figure out where the name comes from, but most of the time it should be okay without. Is it just for clarity, to visually distinguish methods from functions?

(I'm generally fine with running experiments on Newsboat's codebase, so if you're doing this just to see how it goes, it's okay.)

dennisschagt replied:
> Out of curiosity: why all the methods in this PR are called through this?

I guess I added it for clarity.
On rereading, I'm not even sure it improves anything. I removed all of the added `this->` parts for consistency with existing code.

Minoru commented on src/pbview.cpp:
```diff
@@ -285,8 +285,11 @@ void PbView::handle_resize()
+	colorman.apply_colors(std::bind(&newsboat::Stfl::Form::set, &dllist_form, _1,
+			_2));
+	colorman.apply_colors(std::bind(&newsboat::Stfl::Form::set, &help_form, _1,
+			_2));
```
I'm okay with keeping `std::bind` here, but please avoid it in the future as it has [plenty of pitfalls](https://abseil.io/tips/108). Lambdas are usually sufficient.

dennisschagt replied:
I wasn't aware of those pitfalls, thanks, thanks!

Minoru reviewed (approved):
> I did remove the FormAction::get_form() method to reduce the number of classes with access to Stfl::Form::run().

IMHO that's sufficient for now, and the best we can do at the moment actually.

> PbView uses Stfl::Form directly, without using the FormAction class.

Ah, I knew I'm missing something. Thanks!

Feel free to merge this after updating the changelog.

dennisschagt commented on doc/cmdline-commands.dsv:
```diff
@@ -6,6 +6,5 @@ tag||<tagname>||Only display feeds with the tag <tagname>.||tag news
 goto||<case-insensitive substring>||Search for a feed whose name contains the case-insensitive substring.||goto foo
 source||<filename> [...]||Load the specified configuration files. This allows it to load alternative configuration files or reload already loaded configuration files on-the-fly from the filesystem.||source ~/.newsboat/colors
 dumpconfig||<filename>||Save current internal state of configuration to file, so that it can be instantly reused as configuration file.||dumpconfig ~/.newsboat/config.saved
-dumpform||||Dump current dialog to text file. This is meant for debugging purposes only.||dumpform
```
Done: https://github.com/newsboat/newsboat/pull/1403/commits/0523485b6c87820f843cd1e40f434765a62d5ec8

dennisschagt commented:
Rebased onto master and squashed the fixups.

Thank you for the review!

> Feel free to merge this after updating the changelog.

I will merge when CI finishes successfully.
